validation/cubed_sphere_surface_gravity_waves/cubed_sphere_surface_gravity_waves.jl

```diff
@@ -0,0 +1,149 @@
+using Statistics
+using Logging
+using Printf
+using DataDeps
+using JLD2
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.CubedSpheres
+using Oceananigans.Coriolis
+using Oceananigans.Models.HydrostaticFreeSurfaceModels
+using Oceananigans.TurbulenceClosures
+
+using Oceananigans.Diagnostics: accurate_cell_advection_timescale
+
+#####
+##### Progress monitor
+#####
+
+mutable struct Progress
+    interval_start_time :: Float64
+end
+
+function (p::Progress)(sim)
+    wall_time = (time_ns() - p.interval_start_time) * 1e-9
+
+    @info @sprintf("Time: %s, iteration: %d, max(|u⃗|): (%.2e, %.2e) m/s, extrema(η): (min=%.2e, max=%.2e), CFL: %.2e, wall time: %s",
+                   prettytime(sim.model.clock.time),
+                   sim.model.clock.iteration,
+                   maximum(abs, sim.model.velocities.u),
+                   maximum(abs, sim.model.velocities.v),
+                   minimum(sim.model.free_surface.η),
+                   maximum(sim.model.free_surface.η),
+                   sim.parameters.cfl(sim.model),
+                   prettytime(wall_time))
+
+    p.interval_start_time = time_ns()
+
+    return nothing
+end
+
+#####
+##### Script starts here
+#####
+
+ENV["DATADEPS_ALWAYS_ACCEPT"] = "true"
+
+Logging.global_logger(OceananigansLogger())
+
+dd = DataDep("cubed_sphere_32_grid",
+    "Conformal cubed sphere grid with 32×32 grid points on each face",
+    "https://github.com/CliMA/OceananigansArtifacts.jl/raw/main/cubed_sphere_grids/cubed_sphere_32_grid.jld2",
+    "b1dafe4f9142c59a2166458a2def743cd45b20a4ed3a1ae84ad3a530e1eff538" # sha256sum
+)
+
+DataDeps.register(dd)
+
+cs32_filepath = datadep"cubed_sphere_32_grid/cubed_sphere_32_grid.jld2"
+
+# Central (λ, φ) for each face of the cubed sphere
+central_longitude = (0, 90,  0, 180, -90,   0)
+central_latitude  = (0,  0, 90,   0,   0, -90)
+
+function cubed_sphere_surface_gravity_waves(; face_number)
+
+    H = 4kilometers
+    grid = ConformalCubedSphereGrid(cs32_filepath, Nz=1, z=(-H, 0))
+
+    ## Model setup
+
+    model = HydrostaticFreeSurfaceModel(
+              architecture = CPU(),
+                      grid = grid,
+        momentum_advection = VectorInvariant(),
```

Comment:
@ali-ramadhan do we want inertial terms in this setup? We could just do the linear problem for this example?

Reply:
Ah I think I was just testing it but will change to `momentum_advection = nothing` for now, at least until `VectorInvariant()` is fixed for the cubed sphere.